pkg/adaptor/hypervisor/aws/types.go

```diff
@@ -19,3 +24,35 @@ type Config struct {
 func (c Config) Redact() Config {
 	return *util.RedactStruct(&c, "AccessKeyId", "SecretKey").(*Config)
 }
+
+type MetadataRetriever struct {
+	client *imds.Client
+	mac    string // save it as it commonly used in path
+}
+
+func NewMetadataRetriever() *MetadataRetriever {
+	var r MetadataRetriever
+	r.client = imds.New(imds.Options{ClientEnableState: imds.ClientDefaultEnableState}) // use imds.ClientEnabled to enforce enabling
+	mac, err := r.Get("mac")
+	if err != nil {
+		// print warn // will fail on Get
```

Comment:
Better to log the error to help with debugging.

Reply:
right, i missed that, thanks!